src/dataflow/src/logging/timely.rs

```diff
@@ -374,6 +415,52 @@ pub fn construct<A: Allocate>(
                 }
             });
 
+        use differential_dataflow::operators::Count;
+        let messages_sent = thin_collection(messages_sent.as_collection(), delay, |c| {
```

Comment:
I think all of this logic could use a few more comments. the two questions that jump out at me are:

1. why are sent and received two different collections? and
2. why are we thinning the collection?

Reply:
ok my read on the answers for both is

1. it was simpler this way
2. to retain message counts for only the set of channels we are currently tracking?

I think thats all totally fine. it does seem to me like if we kept a is_sent variable around in the messages collection we could do the thinning with only one feedback loop / semijoin instead of two but idt thats worth optimizing for at this stage

Reply:
I think the answer is much more complicated, actually. The `thin` operation is only correct if the thing driving the thinning is correctly serialized with updates being thinned. I can't actually tell if this is happening here, but if the semijoin key is `(channel_id, worker)` then it is important that this record emerge before the thing it is semijoined with. Sent and received messages get logged at different workers (the sender and receiver, resp) and it is important to use their worker IDs.

@antifuchs does this sound right? Certainly worth a detailed comment if so, to prevent any future incorrect "simplification" of this code.

Reply:
Yep, @frankmcsherry - that's correct. The join keys are different for data sent/received on channels, so we have to keep them separate, only later can we twist them together again.